server/src/test/java/org/elasticsearch/common/lucene/store/BytesReferenceIndexInputTests.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.lucene.store;
+
+import org.elasticsearch.common.bytes.BytesArray;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import static org.hamcrest.Matchers.containsString;
+
+public class BytesReferenceIndexInputTests extends ESIndexInputTestCase {
+    public void testRandomReads() throws IOException {
+        for (int i = 0; i < 100; i++) {
```

Comment:
Does it make sense to randomize the amount of cycles in the loop?

Reply:
Not really, this is just here cos this test is so fast that it makes sense to run it a bunch of times.